RFCs/0035-partiql-window-function.md

```diff
@@ -0,0 +1,1399 @@
+
+```
+<<
+    <stock : {'trade_date': 2022-09-30, 'ticker': 'AMZN', 'price': 113.00}>,
+    <stock : {'trade_date': 2022-09-30, 'ticker': 'AMZN', 'price': 113.00}>
+>>
+```
+
+```sql
+WINDOWED (ORDER BY stock.trade_date) PARTITION AS p AT pos
+```
+The output binding collection is:
+
+```
+<<
+   <
+    stock : {'trade_date': 2022-09-30, 'ticker': 'AMZN', 'price': 113.00},
+    p : [
+          <stock : {'trade_date': 2022-09-30, 'ticker': 'AMZN', 'price': 113.00}>,
+          <stock : {'trade_date': 2022-09-30, 'ticker': 'AMZN', 'price': 113.00}>
+        ],
+    pos: 0
+   >,
+   <
+    stock : {'trade_date': 2022-09-30, 'ticker': 'AMZN', 'price': 113.00},
+    p : [
+          <stock : {'trade_date': 2022-09-30, 'ticker': 'AMZN', 'price': 113.00}>,
+          <stock : {'trade_date': 2022-09-30, 'ticker': 'AMZN', 'price': 113.00}>
+        ],
+    pos: 1
+   >
+>>
+```
+**Remark:** 
+
+The position variable `pos` MUST be unique for each binding tuple.
+
+
+#### Operating on WINDOWED Clause Output
+
+The `WINDOWED` clause computes the partition for each row and makes the partition available in its output using partition variable `p` with position variable `pos`. These variables can then get accessed by the next clause (as next clause's input) to perform further operations. The scoping rules and path navigation behave as what has already been specified in the spec's sections `3.4` and `10`. Example 2.1 shows an example of this.
+
+Like other clauses, the output of binding tuples from `WINDOWED` clauses becomes the input of next operator.
+
+Example 2.1: 
+Consider the database: 
+
+$\rho_0$ =
+```
+<
+    stock : <<
+        { 'trade_date' : 2022-09-30, 'ticker' : 'AMZN', 'price' : 113.00 },
+        { 'trade_date' : 2022-10-03, 'ticker' : 'AMZN', 'price' : 115.88 },
+        { 'trade_date' : 2022-09-30, 'ticker' : 'GOOG', 'price' : 96.15 }
+   >>         
+>
+```
+
+If we would like to retrieve the current price and the previous price of a given stock, and if there is no previous data, we want to return null.
+```sql
+SELECT VALUE
+    {'ticker' : stock.ticker,
+     'current_price': stock.price,
+     'previous_price' : CASE WHEN pos - 1 >= 0 THEN CAST ( SELECT VALUE stock.price FROM p AT idx WHERE idx = pos - 1 AS LIST)[0]
+                             ELSE NULL END
+    }
+FROM stock AS stock
+WINDOWED (
+        PARTITION BY stock.ticker
+        ORDER BY stock.trade_date
+    ) PARTITION AS p AT pos
+```
+
+The `WINDOWED` clause outputs a collection of binding tuples(See Example 1.4), which is consumed by the SELECT clause:
+
+B<sup>OUT</sup><sub>WINDOWED</sub> = B<sup>IN</sup><sub>SELECT</sub>  =
+
+```
+<<
+   <
+    stock : {'trade_date': 2022-09-30, 'ticker': 'AMZN', 'price': 113.00},
+    ws_1_partition : [
+          <stock : {'trade_date': 2022-09-30, 'ticker': 'AMZN', 'price': 113.00}>,
+          <stock : {'trade_date': 2022-10-03, 'ticker': 'AMZN', 'price': 115.88}>
+        ],
+    ws_1_pos: 0
```

Comment:
Shouldn't this be `p` and not `ws_1_partition` and `pos` and not `ws_1_pos`?

Reply:
It should. Thanks for point that out and the detailed review.